strategies/affected/src/test/java/org/arquillian/smart/testing/strategies/affected/ClassDependenciesGraphUsingPropertyTest.java

```diff
@@ -0,0 +1,121 @@
+package org.arquillian.smart.testing.strategies.affected;
+
+import java.io.File;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import net.jcip.annotations.NotThreadSafe;
+import org.arquillian.smart.testing.configuration.Configuration;
+import org.arquillian.smart.testing.strategies.affected.fakeproject.main.D;
+import org.arquillian.smart.testing.strategies.affected.fakeproject.test.ATest;
+import org.arquillian.smart.testing.strategies.affected.fakeproject.test.BTest;
+import org.arquillian.smart.testing.strategies.affected.fakeproject.test.CTest;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.contrib.java.lang.system.RestoreSystemProperties;
+import org.junit.experimental.categories.Category;
+import org.junit.rules.TemporaryFolder;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+
+@Category(NotThreadSafe.class)
+public class ClassDependenciesGraphUsingPropertyTest {
+    @Rule
+    public final TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+    @Rule
+    public final RestoreSystemProperties restoreSystemProperties = new RestoreSystemProperties();
+
+    @Test
+    public void should_not_detect_all_changes_transitive_if_transitivity_is_disabled() {
+        // given
+        final AffectedConfiguration affectedConfiguration = new AffectedConfiguration();
+        affectedConfiguration.setTransitivity(false);
+
+        final Configuration configuration = Configuration.load();
+        configuration.setStrategiesConfiguration(Collections.singletonList(affectedConfiguration));
+        configuration.dump(temporaryFolder.getRoot());
+
+        final ClassDependenciesGraph
+            classDependenciesGraph = new ClassDependenciesGraph(new EndingWithTestTestVerifier(), Configuration.load());
```

Comment:
I'm slightly confused here - on line 69, you dump the config to the temporary folder, but here you again read the default values and not from the temp folder...? can be applied for all the following test methods. 
In my opinion, the configuration shouldn't be dumped but passed as a parameter to the `ClassDependenciesGraph`
This opens an interesting question - do you really test anything here if all tests are green with wrong configuration?